src/main/scala/org/biodatageeks/sequila/pileup/converters/common/PileupConverter.scala

```diff
@@ -0,0 +1,39 @@
+package org.biodatageeks.sequila.pileup.converters.common
+
+import org.apache.spark.sql.DataFrame
+import org.biodatageeks.sequila.utils.Columns
+
+trait PileupConverter {
+
+  /**
+   * Loads data from input path and transforms to common format
+   * @param path
+   * @return
+   */
+  def transform (path: String): DataFrame
+
+  /**
+   * Transforms input DF to common format (per-base output and string values for alts and quals)
+   * @param df
+   * @param caseSensitive
+   * @return
+   */
+  def toCommonFormat(df:DataFrame, caseSensitive:Boolean): DataFrame
+
+  /**
+   * Transforms map fields into properly formatted and ordered string fields using UDFs
+   * @param df
+   * @return
```

Comment:
Maybe we really should start caring about better code documentation?

Reply:
is this a bad thing?